apps/scratch/src/utilities/student-activity-tracking.ts

```diff
@@ -1,61 +1,76 @@
 import { CrtContextValue } from "../contexts/CrtContext";
 import { isBlockPartOfLargeStack } from "./scratch-block-stack-utils";
-import type { Block, Workspace } from "scratch-blocks";
+import type { Block } from "scratch-blocks";
 import type { WorkspaceChangeEvent } from "../types/scratch-workspace";
 interface TrackMoveParams {
-  workspace: Workspace;
-  blockId?: string;
-  sendRequest?: CrtContextValue["sendRequest"];
-  action?: StudentAction;
+  block: Block;
+  sendRequest: CrtContextValue["sendRequest"];
+  action: StudentAction;
+  solution: Blob;
 }
 
-export type StudentAction = "create" | "move";
+export enum StudentAction {
+  CREATE = "create",
+  MOVE = "move",
+  DELETE = "delete",
+}
+
+const scratchToStudentAction: Record<string, StudentAction | null> = {
+  create: StudentAction.CREATE,
+  move: StudentAction.MOVE,
+  delete: StudentAction.DELETE,
+};
 
-export function shouldTrackActivity(
+export const mapScratchEventTypeToStudentAction = (
+  type: string,
+): StudentAction | null => scratchToStudentAction[type] || null;
+
+export const shouldTrackActivity = (
+  action: StudentAction,
   event: WorkspaceChangeEvent,
   canEditTask: boolean | undefined,
-): boolean | undefined {
-  return (
-    (event.type === "move" || event.type === "create") &&
-    event.recordUndo &&
-    !canEditTask &&
-    !!event.blockId
-  );
-}
+): boolean | undefined =>
+  (action === StudentAction.MOVE || action === StudentAction.CREATE) &&
+  event.recordUndo &&
+  !canEditTask &&
+  !!event.blockId;
 
-function shouldTrackMove(block: Block | null | undefined): boolean {
-  if (!block) {
-    return false;
-  }
-  return isBlockPartOfLargeStack(block);
-}
+const shouldTrackMove = (block: Block | null | undefined): boolean =>
+  !block ? false : isBlockPartOfLargeStack(block);
+
+const getBlockData = (block: Block): Record<string, unknown> => ({
+  blockId: block.id ?? "",
+  parentId: block.getParent?.()?.id ?? null,
+  position: block.getRelativeToSurfaceXY?.() ?? { x: 0, y: 0 },
+  childrenIds: block.getChildren?.().map((child) => child.id) ?? [],
+});
```

Comment:
For this it may be worth having a quick discussion. It is a good starting point but I do not think the data like this will be useful for researchers. Moreover, this data will be event dependent, i.e. we want different information depending on the type.

For appends I think we for sure also need the type of block that is being appended.
I don't think the position will be useful honestly but I don't mind having it. However I do find it weird that it may be undefined and that we default to (0, 0). When can this be undefined? And if it actually can be undefined, I would either drop it altogether (if we decide we do not need it) or then also supply null to indicate the absence of this information instead of a placeholder value.